crnk-core/src/main/java/io/crnk/core/engine/internal/document/mapper/ResourceMapper.java

```diff
@@ -186,7 +186,7 @@ private boolean isNodeIncluded(ResourceField field, JsonNode node) {
 
 	protected boolean isDefaultNodeValue(JsonNode node) {
 		return isNullNodeValue(node) || (node.isObject() || node.isArray()) && node.size() == 0
-				|| node.asText().isEmpty() || node.isNumber() && node.asDouble() == 0d;
+				|| node.isValueNode() && node.asText().isEmpty() || node.isNumber() && node.asDouble() == 0d;
 	}
```

Comment:
is there possibly a more efficient way? I would expect that asText to be rather inefficient for non primitve fields.

Reply:
I believe the intention of the author was to skip the ` node.asText().isEmpty()` for fields that were not primitive. As I believe `node.isValueNode()` will return false for JSON types that are arrays or objects.

Reply:
ok thx!